iceberg/streams/iceberg/schema_name_mapper.cpp

```diff
@@ -0,0 +1,62 @@
+#include "iceberg/streams/iceberg/schema_name_mapper.h"
+
+namespace iceberg {
+namespace {
+
+void Ensure(bool cond, const std::string& msg) {
+  if (!cond) {
+    throw std::runtime_error(msg);
+  }
+}
+
+void InsertOrFail(std::unordered_set<std::string>& set, const std::string& elem) {
+  Ensure(!set.contains(elem), "Names at the same level are not unique");
+  set.insert(elem);
+}
+
+}  // namespace
+
+SchemaNameMapper::SchemaNameMapper(const std::string& json) {
+  doc_.Parse(json.c_str(), json.size());
+  Ensure(!doc_.HasParseError(), "schema.name-mapping.default is not a valid JSON");
+}
+
+SchemaNameMapper::Node SchemaNameMapper::GetRootNode() const { return Node(&doc_); }
+
+SchemaNameMapper::Node::Node(const rapidjson::Value* doc) : doc_(doc) { Validate(); }
```

Comment:
It's not clear why you validate doc on every Node creation. It would be called on every GetRootNode() creation you make lots of times. Would it be not enough to validate entire json document once?